packages/admin-next/dashboard/src/routes/orders/returns-create/components/create-returns/create-returns.tsx

```diff
@@ -0,0 +1,291 @@
+import React, { useEffect, useRef, useState } from "react"
+import { useForm } from "react-hook-form"
+
+import { AdminPostOrdersOrderReturnsReq, Order } from "@medusajs/medusa"
+import { Button, ProgressStatus, ProgressTabs } from "@medusajs/ui"
+import { useTranslation } from "react-i18next"
+import * as zod from "zod"
+
+import {
+  RouteFocusModal,
+  useRouteModal,
+} from "../../../../../components/route-modal"
+import { ItemsTable } from "../items-table"
+import { ReturnsForm } from "./returns-form"
+import { getDbAmount } from "../../../../../lib/money-amount-helpers"
+import { useAdminRequestReturn, useAdminShippingOptions } from "medusa-react"
+import { getErrorMessage } from "@medusajs/admin-ui/ui/src/utils/error-messages.ts"
+
+type CreateReturnsFormProps = {
+  order: Order
+}
+
+enum Tab {
+  ITEMS = "items",
+  DETAILS = "details",
+}
+
+type StepStatus = {
+  [key in Tab]: ProgressStatus
+}
+
+const CreateReturnSchema = zod.object({
+  quantity: zod.record(zod.string(), zod.number()),
+  reason: zod.record(zod.string(), zod.string().optional()),
+  note: zod.record(zod.string(), zod.string().optional()),
+  location: zod.string(),
+  shipping: zod.string(),
+  send_notification: zod.boolean().optional(),
+
+  enable_custom_refund: zod.boolean().optional(),
+  enable_custom_shipping_price: zod.boolean().optional(),
+
+  custom_refund: zod.number().optional(),
+  custom_shipping_price: zod.number().optional(),
+})
+
+export function CreateReturns({ order }: CreateReturnsFormProps) {
+  const { t } = useTranslation()
+  const { handleSuccess } = useRouteModal()
+
+  const [selectedItems, setSelectedItems] = useState([])
+  const [tab, setTab] = React.useState<Tab>(Tab.ITEMS)
+  const [isSubmitting, setIsSubmitting] = useState(false)
+
+  const { mutateAsync: requestReturnOrder } = useAdminRequestReturn(order.id)
+
+  const { shipping_options = [] } = useAdminShippingOptions({
+    region_id: order.region_id,
+    is_return: true,
+  })
+
+  const refundableAmount = useRef(0)
+
+  // TODO: should we filter fullfilled items only here
+  const selected = order.items.filter((i) => selectedItems.includes(i.id))
+
+  const form = useForm<zod.infer<typeof CreateReturnSchema>>({
+    defaultValues: {
+      // Items not selected so we don't know defaults yet
+      quantity: {},
+      reason: {},
+      note: {},
+
+      location: "",
+      shipping: "",
+      send_notification: !order.no_notification,
+
+      enable_custom_refund: false,
+      enable_custom_shipping_price: false,
+
+      custom_refund: 0,
+      custom_shipping_price: 0,
+    },
+  })
+
+  const onSubmit = form.handleSubmit(async (data) => {
+    setIsSubmitting(true)
```

Comment:
you can use `isLoading` from `useAdminRequestReturn` instead of managing this manually